spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRouteLocator.java

```diff
@@ -238,6 +250,13 @@ public void resetRoutes() {
 		return values;
 	}
 
+	protected String mapRouteToService(String serviceId) {
+		if(this.serviceRouteMapper != null) {
```

Comment:
You could use auto-configuration if regex mapper is enabled to create one, otherwise the default would be a passthru rather than an `if` statement here.